File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    queue<pair<int, int>> q;
    for (int i = 0; i< progresses.size(); i++){
        q.push({progresses[i], speeds[i]});
    }
    
    for (int day = 1; day <= 100 && !q.empty(); ++day){
        int cnt = 0;
        
        while (q.front().first + q.front().second * day >= 100){
            q.pop();
            cnt++;
        }
        
        if(cnt) {
            answer.push_back(cnt);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] cnt = new int[100];
        int day = -1;
        
        for (int i = 0; i < progresses.length; ++i) {
            while (progresses[i] + speeds[i] * day < 100) {
                day++;
            }
            cnt[day]++;
        }
        return Arrays.stream(cnt)
            .filter(i -> i != 0)
            .toArray();
    }
}